lib/facility_factory.rb

```diff
@@ -0,0 +1,69 @@
+class FacilityFactory
+  attr_reader :operating_facilities
+
+  def initialize
+    @operating_facilities = []
+  end
+
+  def define_facilities(facilities)
+    facilities.each do |facility_data|
+      if facility_data[:location_1].is_a?(Hash)
+        address_data = JSON.parse(facility_data[:location_1][:human_address], symbolize_names: true)
```

Comment:
This is where helper methods can be useful to shorten up this method to make it more clear what's going on. Could have each block within each flow (if/elsif) be it's own helper method.